src/App.ts

```diff
@@ -631,12 +685,26 @@ export default class App {
     let { client } = this;
     const token = selectToken(context);
     if (token !== undefined) {
-      let pool = this.clients[source.teamId];
-      if (pool === undefined) {
-        // eslint-disable-next-line no-multi-assign
-        pool = this.clients[source.teamId] = new WebClientPool();
+      let pool;
+      const clientOptionsCopy = { ...this.clientOptions };
+      if (source.teamId !== undefined) {
+        pool = this.clients[source.teamId];
+        if (pool === undefined) {
+          // eslint-disable-next-line no-multi-assign
+          pool = this.clients[source.teamId] = new WebClientPool();
+        }
+        clientOptionsCopy.teamId = source.teamId;
+      } else if (source.enterpriseId !== undefined) {
+        pool = this.clients[source.enterpriseId];
+        if (pool === undefined) {
+          // eslint-disable-next-line no-multi-assign
+          pool = this.clients[source.enterpriseId] = new WebClientPool();
+        }
+      }
+      if (pool !== undefined) {
+        // Question: should teamId from source or authResult be passed in via clientOptionsCopy
```

Comment:
In the case of enterprise_install, yes, I think so. We can set the default team_id to the client instance.

Reply:
I decided to update this to use the `teamID` from `authorizeResult` first if it exists, if not use the teamId from `source`